### PR TITLE
fixed sardine to close connection

### PR DESCRIPTION
**Pull request by Tantalor93**

### CODE REVIEW

**Comment by pirkogdc** [comment]
Please consider posting a patch to sardine or at least creating an issue and referncing that issue here.

File: src/main/java/com/gooddata/gdc/GdcSardine.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+package com.gooddata.gdc;
+
+import com.github.sardine.impl.SardineImpl;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import java.io.IOException;
+
+public class GdcSardine extends SardineImpl {
```
Comment (martiner) [inline src/main/java/com/gooddata/gdc/GdcSardine.java]: pls package protected

File: src/main/java/com/gooddata/gdc/GdcSardine.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    /**
+     * had to be overriden, because parent method did not close connection after execution
+     */
+    @Override
+    protected <T> T execute(HttpRequestBase request, ResponseHandler<T> responseHandler) throws IOException {
+        try {
+            return super.execute(request, responseHandler);
+        } finally {
+            request.releaseConnection();
```
Comment (martiner) [inline src/main/java/com/gooddata/gdc/GdcSardine.java]: is request checked for null somewhere? eg. in the super.execute()?

File: src/test/java/com/gooddata/AbstractGoodDataAT.java [hunk 3]
```diff
@@ -17,8 +28,23 @@
     protected static final String title =
             "sdktest " + new LocalDate() + " " + System.getenv("BUILD_NUMBER");
 
+    protected static final AtomicReference<PoolingHttpClientConnectionManager> connManagerRef = new AtomicReference<>();
```
Comment (martiner) [inline src/test/java/com/gooddata/AbstractGoodDataAT.java]: probably doesn't have to be atomic reference

**Pull request merged**